### ISSUE TITLE
Dynamic dependencies

### ISSUE THREAD

**Issue by stephanlachnit** [mesonbuild/imgui#7]
When building just the core lib, we don't need any of the dependencies.

I suggest something like:

```meson
dependencies = []

if get_option('sdl2').enabled()
    sources += 'examples/imgui_impl_sdl.cpp'
    dependencies += dependency('sdl2')
endif
```

The same should be done with the example include dir.
I. e.:
```meson
main_include_dir = include_directories('.')
example_include_dir = include_directories('example')
```
and make the inclusion dependent on the options. However this is more complicated and less important than the first point.

**Comment by eli-schwartz** [comment]
https://github.com/mesonbuild/imgui/blob/0f33384098394c515c6e83fe1e1e27d39e0c3b5d/meson.build#L58-L66

It should probably try to find each one using feature options, and if successfully found, add it to an array of dependencies, then append the relevant source file... rather than inlining it into the `library()` call.

```
sdl_dep = dependency('sdl2', required: get_option('sdl2'))
sdl_dep.found()
```
should short-circuit if `-Dsdl2=disabled`, but currently if it is `auto`, the dependency gets non-fatally looked for and the sources are not included.

That being said, it seems to me like it won't ever force you to have the dependency installed when the meson option isn't enabled...

**Comment by eli-schwartz** [comment]
Fixed in #8 but github did not close it correctly...
